dm/dm-continuous-data-validation.md

```diff
@@ -0,0 +1,285 @@
+
+    ```
+    Usage:
+      dmctl validation clear-error <task-name> <error-id|--all> [flags]
+
+    Flags:
+          --all    all errors
+      -h, --help   help for clear-error
+    ```
+
+- `ignore-error`: ignore the error row. This error row is marked as "ignored".
+
+    ```
+    Usage:
+      dmctl validation ignore-error <task-name> <error-id|--all> [flags]
+
+    Flags:
+          --all    all errors
+      -h, --help   help for ignore-error
+    ```
+
+- `resolve-error`: the error row is manually handled and marked as "resolved".
+
+    ```
+    Usage:
+      dmctl validation resolve-error <task-name> <error-id|--all> [flags]
+
+    Flags:
+          --all    all errors
+      -h, --help   help for resolve-error
+    ```
+
+## Stop continuous data validation
+
+To stop the continuous data validation, run the `validation stop` command:
+
+```
+Usage:
+  dmctl validation stop [--all-task] [task-name] [flags]
+
+Flags:
+      --all-task   whether applied to all tasks
+  -h, --help       help for stop
+```
+
+For detailed usage, refer to [`dmctl validation start`](#method-2-enable-using-dmctl).
+
+## Implementation
+
+The architecture of continuous data validation (validator) in DM is as follows:
+
+![validator summary](/media/dm/dm-validator-summary.jpeg)
+
+The lifecycle of continuous data validation is as follows:
+
+![validator lifecycle](/media/dm/dm-validator-lifecycle.jpeg)
+
+The detailed implementation of continuous data validation is as follows:
+
+1. The validator pulls a binlog event from the upstream and gets the changed rows:
+    - The validator only checks a event that has been incrementally migrated by the syncer. If the event has not been processed by the syncer, the validator pauses and waits for the syncer to complete processing.
+    - If the event has been processed by the syncer, the validator moves on to the following steps.
+2. The validator parses the binlog event and filters out the rows based on the block and allow lists, the table filters, and table routing. After that, the validator submits the changed rows to the validation worker that runs in the background.
+3. The validation worker merges the changed rows that affect the same table and the same primary key to avoid validating "expired" data. The changed rows are cached in memory.
+4. When the validation worker has accumulated a certain number of changed rows or when a certain time interval is passed, the validation worker queries the downstream database using the primary keys to get the current data and compares it with the changed rows.
+5. The validation worker performs the data validation. If the validation mode is `full`, the validation worker compares data of the changed rows with data of the downstream database. if the validation mode is `fast`, the validation worker only checks the existence of the changed rows.
+    - If the validation succeeds, the changed row is removed from the memory.
```

Comment:
```suggestion
    - If the changed rows pass the validation, the changed row is removed from the memory.
```
"validation succeeds" sounds a bit elliptical or ambiguous, and sounds more likely to be a validation procedure without encountering hitches.